src/panoramic/cli/data_connections.py

```diff
@@ -0,0 +1,155 @@
+    """CLI command. Update specific data connection."""
+    data_connections = DataConnections.load()
+    if name not in data_connections:
+        raise click.ClickException(f'Data connection with name "{name}" not found.')
+
+    if password_stdin:
+        password = click.prompt('Enter password: ', hide_input=True, type=str)
+
+    if type:
+        data_connections[name]['type'] = type
+    if user:
+        data_connections[name]['user'] = user
+    if password:
+        data_connections[name]['password'] = password
+    if host:
+        data_connections[name]['host'] = host
+    if port:
+        data_connections[name]['port'] = port
+    if database_name:
+        data_connections[name]['database_name'] = database_name
+
+    if not no_test:
+        ok, error = DataConnections.test(data_connections[name])
+        if not ok:
+            print(f'Failed to create data connection: {error}')
+            return
+
+    DataConnections.save(data_connections)
+    print('Data connection was successfully created!')
+
+
+def list_data_connections_command(show_password):
+    """CLI command. List all data connections."""
+    data_connections = DataConnections.load()
+    if not data_connections:
+        config_file = Paths.config_file()
+        print(
+            f'No data connections found.\n'
+            f'Use "pano data-connections create" to create data connection or edit "{config_file}" file.'
+        )
+        exit(0)
+
+    for name, connection in data_connections.items():
+        connection_string = DataConnections.create_connection_string(connection)
+        if not show_password:
+            connection_string = connection_string.replace(connection['password'], '*****')
+        print(f'{name}: {connection_string}')
+
+
+def remove_data_connection_command(name):
+    """CLI command. Remove existing data connection."""
+    data_connections = DataConnections.load()
+    if name not in data_connections:
+        raise click.ClickException(f'Data connection with name "{name}" not found.')
+
+    del data_connections[name]
+    DataConnections.save(data_connections)
+
+
+def test_all_data_connections_command():
+    """CLI command. Test all data connections by trying to connect to the database."""
+    for name, connection in DataConnections.load().items():
+        ok, error = DataConnections.test(connection)
+        if ok:
+            print(f'{name}... OK')
+        else:
+            print(f'{name}... FAIL: {error}')
+
+
+class DataConnections:
+    def __init__(self):
+        self._connections = self.load()
+
+    def get_by_name_cached(self, name: str) -> Dict[str, str]:
+        """Get data connection by name. Use when checking data connections very often, like in for loop.
+        The cached data connections will be used instead of loading data connections file every time."""
+        # TODO: what we want to do when data connection with given name doesnt exist? raise Exception or return None...?
```

Comment:
I'd say raising sth like `ConnectionNotFound` is okay as long as we appropriately handle it in the logic above.

Reply:
added exception